File: android/src/main/java/kh/ad/reve_chat_sdk/UserModel.java
package kh.ad.reve_chat_sdk;

import java.util.HashMap;

public class UserModel {
    private final String email;
    private final String name;
    private final String phoneNumber;

    private UserModel(String email, String name, String phoneNumber) {
        this.email = email;
        this.name = name;
        this.phoneNumber = phoneNumber;
    }

    public String getEmail() {
        return email;
    }

    public String getName() {
        return name;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    static UserModel fromMap(HashMap<String, ?> json) {
        String name = (String) json.get("username");
        String email = (String) json.get("userEmail");
        String phoneNumber = (String) json.get("phoneNumber");
        return new UserModel(
                email,
                name,
                phoneNumber
        );
    }
}

File: ios/Classes/UserModel.swift
import Foundation
class UserModel
{
    private var email : String;
    private var name : String;
    private var phoneNumber : String;
    init(_ email : String, _ name : String, _ phoneNumber : String)
    {
        self.email = email;
        self.name = name;
        self.phoneNumber = phoneNumber;
    }
    func getEmail() -> String
    {
        return self.email;
    }
    func getName() -> String
    {
        return self.name;
    }
    func getPhoneNumber() -> String
    {
        return self.phoneNumber;
    }
    static func fromMap(_ json : [String : Any]) -> UserModel
    {
        let name : String = json["username"] as! String;
        let email : String = json["userEmail"] as! String;
        let phoneNumber : String = json["phoneNumber"] as! String;
        return UserModel(email, name, phoneNumber);
    }
}
